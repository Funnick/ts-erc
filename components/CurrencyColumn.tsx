import { Column, Text, CurrencyWrapper,
    FlagImage, Dropdown, DropOption } from './styled/styledComponents';

type CurrencyProps = {
    title: string
    flagCode: string
    currency: string
    setCurrency: (value: string) => void
    currencies: string[]
}

const CurrencyColumn = ({ title, flagCode, currency, setCurrency, currencies }: CurrencyProps) => {
return (
    <Column>
        <Text>{title}</Text>

        <CurrencyWrapper>
            <FlagImage
                src={`https://flagsapi.com/${flagCode}/flat/64.png`}
                height={25}
                width={25}
                alt={`${flagCode}-flag`}
            />
            <Dropdown value={currency} onChange={e => setCurrency(e.target.value)}>
                {currencies.map(curr => {
                    return <DropOption key={curr} value={curr}>{curr}</DropOption>
                })}
            </Dropdown>
        </CurrencyWrapper>
    </Column>
)}

export default CurrencyColumn