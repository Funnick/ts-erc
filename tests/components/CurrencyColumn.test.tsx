import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import CurrencyColumn from '@/components/CurrencyColumn';

describe('Component Test CurrencyColumn', () => {
    it('Should render a currency column properly', () => {
        render(<CurrencyColumn title="From" 
        flagCode="US" currency="USD"
        setCurrency={(s: string) => {}}
        currencies={["EUR", "CAD", "USD"]} />)
        
        const text = screen.getByText('From')
        const dropdown = screen.getByRole('combobox')
        const options = screen.getAllByRole('option')
        
        expect(text).toBeInTheDocument()
        expect(dropdown).toBeInTheDocument()
        expect(options.length).toBe(3)
        expect(options[0]).toHaveTextContent("EUR")
        expect(options[1]).toHaveTextContent("CAD")
        expect(options[2]).toHaveTextContent("USD")
    })

    it('Snapshot', () => {
        const tree = renderer.create(
        <CurrencyColumn title="From" 
        flagCode="US" currency="USD"
        setCurrency={(s: string) => {}}
        currencies={["EUR", "CAD", "USD"]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
}) 