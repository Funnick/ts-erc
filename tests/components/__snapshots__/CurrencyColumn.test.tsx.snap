// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test CurrencyColumn Snapshot 1`] = `
<div
  className="sc-ispOId duzTmJ"
>
  <p
    className="sc-fFGjHI fxeOjw"
  >
    From
  </p>
  <div
    className="sc-kcuKUB cgcjto"
  >
    <img
      alt="US-flag"
      className="sc-dcrmVg deDNXi"
      height={25}
      src="https://flagsapi.com/US/flat/64.png"
      width={25}
    />
    <select
      className="sc-kuWgmH dWoOPP"
      onChange={[Function]}
      value="USD"
    >
      <option
        className="sc-duSInm"
        value="EUR"
      >
        EUR
      </option>
      <option
        className="sc-duSInm"
        value="CAD"
      >
        CAD
      </option>
      <option
        className="sc-duSInm"
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
</div>
`;
